libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/ApolloGenerateUsedCoordinatesAggregateTask.kt

```diff
@@ -0,0 +1,33 @@
+package com.apollographql.apollo3.gradle.internal
+
+import com.apollographql.apollo3.compiler.toUsedCoordinates
+import com.apollographql.apollo3.compiler.writeTo
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+
+@CacheableTask
+abstract class ApolloGenerateUsedCoordinatesAggregateTask : DefaultTask() {
+  @get:OutputFile
+  @get:Optional
+  abstract val outputFile: RegularFileProperty
+
+  @get:InputFiles
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  abstract val incomingUsedCoordinates: ConfigurableFileCollection
+
+  @TaskAction
+  fun taskAction() {
+    val coordinates = incomingUsedCoordinates.files.fold(setOf<String>()) { acc, n -> acc + n.toUsedCoordinates() }
```

Comment:
It's not.  `val coordinates = incomingUsedCoordinates.files.map { it.toUsedCoordinates() }` returns a `List<Set<String>>` and not a `Set<String>` like the "fold" version. Let's make a puzzler about it? 😅

Reply:
oooh right! So it's the same as `incomingUsedCoordinates.files.flatMap{ it.toUsedCoordinates() }.toSet()`, right? 😁 (I'm not saying it's clearer than `fold`)

Reply:
Yup, I think that'd work. Although if there are a lot duplicate coordinates, the fold version will (most likely?) allocate a bit less because it builds the set "greedily" while the `flatMap` creates a big list first. Maybe I'm wrong though. Just my hunch about it...